lib/utils/config.ts

```diff
@@ -66,27 +67,30 @@ export function generateBaseConfig(
 
 export function generateApplicationConfig(
 	application: ResinSdk.Application,
-	options: {},
+	options: { version?: string },
 ) {
-	return generateBaseConfig(application, options).tap(config =>
-		addApplicationKey(config, application.id),
-	);
+	return generateBaseConfig(application, options).tap(config => {
+		if (semver.satisfies(options.version, '>=2.7.8')) {
+			return addProvisioningKey(config, application.id);
+		} else {
+			return addApplicationKey(config, application.id);
+		}
+	});
 }
 
 export function generateDeviceConfig(
-	device: ResinSdk.Device & { application_name: string },
+	device: ResinSdk.Device & { belongs_to__application: ResinSdk.PineDeferred },
 	deviceApiKey: string | true | null,
-	options: {},
+	options: { version?: string },
 ) {
 	return resin.models.application
-		.get(device.application_name)
+		.get(device.belongs_to__application.__id)
```

Comment:
I guess that we should eventually move those pine helpers in the SDK (and later in pine itself).